lib/trino-orc/src/main/java/io/trino/orc/reader/SliceDictionaryColumnReader.java

```diff
@@ -219,6 +219,7 @@ private void setDictionaryBlockData(byte[] dictionaryData, int[] dictionaryOffse
             dictionaryOffsets[positionCount] = dictionaryOffsets[positionCount - 1];
             dictionaryBlock = new VariableWidthBlock(positionCount, wrappedBuffer(dictionaryData), dictionaryOffsets, Optional.of(isNullVector));
             currentDictionaryData = dictionaryData;
+            systemMemoryContext.setBytes(getRetainedSizeInBytes());
```

Comment:
nit: see https://github.com/trinodb/trino/pull/10475 which will actually make Table scan block in case of tight memory situation